unit-tests/to_upper_lower.F90

```diff
@@ -0,0 +1,48 @@
+!> @file
+
+!> output
+!!
+!! Unit test for to_upper() and to_lower() functions under UFS_UTILS package
+!! exit code other than zero identify issue found
+!!
+ 
+program to_upper_lower
+
+ 
+ implicit none
+
+ logical                          :: match_result
+
+ character(len=12)                :: test_input_char_1, test_input_char_2, u_st_base, l_st_base
+
+ u_st_base="STAGGERLOCCE"
+ l_st_base="staggerlocce"
+ test_input_char_1="sTAGGErLOCCE"
+ test_input_char_2="staGGErLOCCE"
+
+ print*, "-------- Starting Unit Testing to_upper_lower  ----------"
+ print*, "------------ Testing to_lower and to_upper --------------"
+
+!-------------------------------------------------------------------------
+! Execute testing below by running target function with testing string 
+!   When match_result set to be T - compare to base line is identical
+!   When match_result set to be F - compare to base line is NOT identical
+!-------------------------------------------------------------------------
+
+ call to_lower(test_input_char_1) 
+ match_result = test_input_char_1 == l_st_base
```

Comment:
If this is not correct, fail immediately. In general we want to stop a test as soon as we have found a problem. You can just use a stop here...